shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -340,7 +340,11 @@ - (instancetype)initWithEngine:(nonnull FlutterEngine*)engine
     }
     _engine = engine;
     CommonInit(self);
-    [engine setViewController:self];
+    if (engine.running) {
+      [self loadView];
+      engine.viewController = self;
```

Comment:
I'm not sure if we'd like to keep this behavior for multi-window. Once Flutter supports multi-window, creating a FVC with an existing engine (that has a view controller already) should probably _add_ a new view to the engine instead of _replacing_ the existing view.

I think we should assert that the engine doesn't have a view controller yet, which will be lifted after we support multi-window.

Reply:
I replaced NSLog with NSAssert.